src/common/snippets/src/op/subgraph.cpp

```diff
@@ -296,6 +316,12 @@ ov::PartialShape snippets::op::Subgraph::canonicalize(const BlockedShapeVector&
     // to align precision inside Subgraph body that is supported by Plugin
     align_element_types(outputShapes, inputShapes);
 
+    // todo: we need a slightly more general approach for backward ROI propagation
+    const auto& result_parent = body_results[0]->get_input_node_shared_ptr(0);
+    if (body_results.size() == 1 &&
+        ov::is_type<opset1::Transpose>(result_parent) &&
+        ov::is_type<snippets::op::Brgemm>(result_parent->get_input_node_shared_ptr(0)))
+        outPShape = result_parent->get_input_partial_shape(0);
```

Comment:
Shouldn't we just skip outPShape computation above in case config.m_has_domain_sensitive_ops == true?

Reply:
Yes, let's disable them, but not for all `config.m_has_domain_sensitive_ops == true` cases, but only for the condition you highlighted. It would be safer to throw an exception if an unsupported case is found rather than silently return invalid master shape.